pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -391,6 +407,13 @@ func applyUpstreamTLSSettings(cluster *v2.Cluster, tls *networking.TLSSettings)
 			},
 			Sni: tls.Sni,
 		}
+		if cluster.Http2ProtocolOptions != nil {
+			// This is HTTP/2 in-mesh cluster, advertise it with ALPN.
+			cluster.TlsContext.CommonTlsContext.AlpnProtocols = ALPNInMeshH2
+		} else {
+			// This is in-mesh cluster, advertise it with ALPN.
+			cluster.TlsContext.CommonTlsContext.AlpnProtocols = ALPNInMesh
```

Comment:
Will this fail to negotiate h1.1?

Reply:
You need to wait for @incfly 's PR to land. MUTUAL does not mean the other end is Istio mTLS. I would actually suggest merging this PR with #5418